### PR TITLE
Automation: Allow spider to run if no OK response

### PR DESCRIPTION
**Pull request by psiinon**
Fixes https://github.com/zaproxy/zaproxy/issues/7510

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: addOns/automation/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Changed
 - Maintenance changes.
 
+### Fixed
+- Allow spider to run if no OK response.
```
Comment (thc202) [inline addOns/automation/CHANGELOG.md:11]: `… response (Issue 7510).`

File: addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/SpiderJobUnitTest.java [hunk 2]
```diff
@@ -536,7 +536,7 @@ void shouldFailIfInvalidHost() throws Exception {
     }
 
     @Test
-    void shouldFailIfForbiddenResponse() throws Exception {
+    void shouldWarnIfForbiddenResponse() throws Exception {
```
Comment (thc202) [inline addOns/automation/src/test/java/org/zaproxy/addon/automation/jobs/SpiderJobUnitTest.java:539]: `Forbidden` → `NotOk`

**Comment by thc202** [comment]
This needs to be changed in the Spider add-on too.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**